### PR TITLE
Use std::make_shared/unique instead of plain 'new'.

### PR DESCRIPTION
**Pull request by bangerth**
Part of using C++11 better, see #2375.

### CODE REVIEW

File: source/simulator/core.cc [hunk 1]
```diff
@@ -143,10 +144,14 @@ namespace aspect
...
                              ParameterHandler &prm)
     :
-    assemblers (new Assemblers::Manager<dim>()),
+    assemblers (std_cxx14::make_unique<Assemblers::Manager<dim>>()),
     parameters (prm, mpi_communicator_),
-    melt_handler (parameters.include_melt_transport ? new MeltHandler<dim> (prm) : nullptr),
-    newton_handler (parameters.nonlinear_solver == NonlinearSolver::iterated_Advection_and_Newton_Stokes ? new NewtonHandler<dim> () : nullptr),
+    melt_handler (parameters.include_melt_transport ?
+                  std_cxx14::make_unique<MeltHandler<dim>>(prm) :
+                  std::unique_ptr<MeltHandler<dim>>()),
```
Comment (tjhei) [inline source/simulator/core.cc]: can you use ``  ... : nullptr`` here? If yes, wouldn't that be more clear and more concise?

**Comment by bangerth** [comment]
Like this? I was concerned that using different types in the `?:` expression would make things ambiguous, but it seems to compile for me.

**Comment by tjhei** [comment]
> expression would make things ambiguous

``nullptr`` is convertible to a ``unique_ptr<...>`` so the resulting type should be the common type of the two args.

**Comment by tjhei** [comment]
> But `std::unique_ptr` is also convertible to a regular pointer, isn't it?

But it can not be converted to ``nullptr_t``, while the second arg can be converted to the first type.

**Comment by bangerth** [comment]
Excellent point!

**Pull request merged**